shared/github_util.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+//go:generate mockgen -destination sharedtest/github_util_mock.go -package sharedtest github.com/web-platform-tests/wpt.fyi/shared GitHubUtil
+
+package shared
+
+import (
+	"context"
+
+	"github.com/google/go-github/v31/github"
+)
+
+const sourceOwner string = "web-platform-tests"
+const sourceRepo string = "wpt-metadata"
+const baseBranch string = "master"
+
+// GitHubUtil encapsulates the implementation of accessing the information in the wpt-metadata repo.
+type GitHubUtil interface {
```

Comment:
I think this interface should probably be moved into `metadata_util.go`, named as `MetadataUtil`, and include the existing `CollectMetadataWithURL` function, too. These two functions are both utility functions for the `wpt-metadata` repo so it's weird to have them scattered in different places.